lib/screens/authentication/sign_up2/mappers/claim_invite_mapper.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:seeds/datasource/remote/model/invite_model.dart';
+import 'package:seeds/domain-shared/page_command.dart';
+import 'package:seeds/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/i18n/authentication/sign_up/sign_up.i18n.dart';
+import 'package:seeds/screens/authentication/sign_up2/viewmodels/page_commands.dart';
+import 'package:seeds/screens/authentication/sign_up2/viewmodels/signup_bloc.dart';
+import 'package:seeds/utils/string_extension.dart';
+
+class ClaimInviteMapper extends StateMapper {
+  SignupState mapValidateInviteCodeToState(SignupState currentState, Result result) {
+    if (result.isError) {
+      return currentState.copyWith(
+        claimInviteView: ClaimInviteView.fail, // gray screen
+        pageCommand: ShowErrorMessage(''), // dialog
+        errorMessage: 'No invites found, try another code'.i18n,
+      );
+    } else {
+      final InviteModel inviteModel = result.asValue!.value;
+
+      if (!inviteModel.isClaimed) {
+        // invite code is valid and can be claimed
+        return currentState.copyWith(claimInviteView: ClaimInviteView.success, inviteModel: inviteModel);
+      } else {
+        // invite code is valid but claimed before
+        return currentState.copyWith(
+          claimInviteView: ClaimInviteView.fail, // gray screen
+          pageCommand: ShowErrorMessage(''), // dialog
```

Comment:
no error message?

Reply:
nop but there is a dialog see comment next the that instruction.